WooCommerce/Classes/ViewRelated/Coupons/CouponDetails/CouponDetails.swift

```diff
@@ -24,59 +50,96 @@ struct CouponDetails: View {
         GeometryReader { geometry in
             ScrollView {
                 VStack(alignment: .leading, spacing: 0) {
-                    Text(Localization.performance)
-                        .bold()
-                        .padding(Constants.margin)
-                        .padding(.horizontal, insets: geometry.safeAreaInsets)
-                    HStack(spacing: 0) {
-                        VStack(alignment: .leading, spacing: Constants.verticalSpacing) {
-                            Text(Localization.discountedOrders)
-                                .secondaryBodyStyle()
-                            Text(viewModel.discountedOrdersCount)
-                                .font(.title)
+                    // Anchor the action sheet at the top to be able to show the popover on iPad in the most appropriate position
+                    Divider()
+                        .actionSheet(isPresented: $showingActionSheet) {
+                            ActionSheet(
+                                title: Text(Localization.manageCoupon),
+                                buttons: [
+                                    .default(Text(Localization.copyCode), action: {
+                                        UIPasteboard.general.string = viewModel.couponCode
+                                        let notice = Notice(title: Localization.couponCopied, feedbackType: .success)
+                                        noticePresenter.enqueue(notice: notice)
+                                    }),
+                                    .default(Text(Localization.shareCoupon), action: {
+                                        showingShareSheet = true
+                                    }),
+                                    .cancel()
+                                ]
+                            )
+                        }
+                        .shareSheet(isPresented: $showingShareSheet) {
+                            ShareSheet(activityItems: [viewModel.shareMessage])
```

Comment:
I'm not quite happy with the ShareSheet here, since it wraps `UIActivityViewController` and displays it full screen while displaying the controller directly shows the small sheet nicely. 

In order to use the controller, we'll need to add a closure and let the hosting controller take care of presenting the share controller. But this didn't work out on iPad since it requires a `sourceView`, which the hosting controller has no reference to without a hacky workaround. So I'm keeping the ShareSheet - let me know if you have any suggestion to improve it!